### ISSUE TITLE
Text-line Detection

### ISSUE THREAD

**Issue by ghost** [dhlab-epfl/dhSegment#16]
@SeguinBe @solivr Thank you for your hard work.

Regarding training a text-line detector from scratch,
1. Since I'm only interested in text-lines; Can I only color the text-lines as boxes in **`Red`**  and make the **`classes.txt`** file as:
```
0 0 0
255 0 0
```

2. What is the suitable `config.json` to be used for training a text-line detection from scratch, since [**`demo_config.json`**](https://github.com/dhlab-epfl/dhSegment/blob/master/demo/demo_config.json#L16) requires a pretrained_model named `resnet50`, which is not for line detection.
```
python train.py with demo/demo_config.json
```

3. @solivr you suggested in [**`issue 9`**](https://github.com/dhlab-epfl/dhSegment/issues/9#issuecomment-389916889) to use the [**`cBAD dataset`**](https://zenodo.org/record/257972) for baseline detection, I have downloaded this dataset, and noticed that the annotations are in **`page xml`**, so how can I use this dataset with it's page xml in training?

![22-compressed-86](https://user-images.githubusercontent.com/33478216/46514395-6c0fe300-c866-11e8-88ac-a0ebced2ba6e.jpg)

4. You have mentioned in your github main-page that you have used the [**`annotator1 from pagenet`**](https://github.com/ctensmeyer/pagenet/blob/master/annotations/cbad_train_annotator_1.txt), how did you use it, if it's in x1, y1, x2, y2, x3, y3, x4, y4 format?

5. You also have mentioned that the training images have been downsized to have 1M pixels each, **`does downsizing the training images reduce the recognition quality?`**

**Comment by SeguinBe** [comment]
Most of your questions have their answer in the [accompanying publication](https://arxiv.org/abs/1804.10371). We always convert what we want to detect as an image of labels, which is a task dependent process. Some code for this can be found in relevant folders in the `exps` directory.

Pretrained models are widely seen as a good initialization, even if they were trained for something different.

The image are downscaled because the network only sees a local context of a fixed number of pixels. Scans are often very high resolution which makes the local context uninformative. In short, downsizing is important to attain good performance, and it is done on the fly by `dh_segment`.

**Comment by solivr** [comment]
Some additional details to your questions : 
1. Yes you can.
2. The `pretrained_model` is used for the encoder part of the network (as @SeguinBe suggested, have a look at the paper), so you'll need it for any task if you want to be efficient. 
Regarding the `config.json` you can copy the `demo_config.json` and update the `training_params`  like this : 
    ```
    "batch_size": 16,
    "input_resized_size": 1000000,
    "make_patches": true,
    "patch_shape": [
        300,
        300
    ],
    "training_margin": 16
    ```
3. For each image, you need to generate the corresponding label image. You can have a look at an example here : https://github.com/dhlab-epfl/dhSegment/blob/master/exps/cbad/utils.py
4. Same as point 3 you can look at https://github.com/dhlab-epfl/dhSegment/blob/master/exps/page/utils.py

**Comment by ghost** [comment]
Thank you both, your amazing!!
